controllers/spiaccesstoken_controller.go

```diff
@@ -322,6 +336,48 @@ func (r *SPIAccessTokenReconciler) oAuthUrlFor(ctx context.Context, at *api.SPIA
 	return oauthBaseUrl + "?state=" + state, nil
 }
 
+func (r *SPIAccessTokenReconciler) refreshToken(ctx context.Context, at *api.SPIAccessToken, sp serviceprovider.ServiceProvider) error {
+	lg := log.FromContext(ctx)
+	token, err := r.TokenStorage.Get(ctx, at)
+	if err != nil {
+		return fmt.Errorf("unable to get refresh token from storage: %w", err)
+	}
+
+	configs, err := r.ServiceProviderFactory.GetAllServiceProviderConfigs(ctx, at.Namespace)
+	if err != nil {
+		return fmt.Errorf("failed to get all known service provider configurations: %w", err)
+	}
+
+	var clientId, clientSecret string
+	for _, conf := range configs {
+		if strings.TrimPrefix(conf.ServiceProviderBaseUrl, "https://") == strings.TrimPrefix(at.Spec.ServiceProviderUrl, "https://") {
```

Comment:
We agreed with Michal that there should be a method for searching the correct configuration instead of doing it here. Moreover, it looks like we are filtering the config in multiple places but always based on some different parameters. That's why I created `GetAllServiceProviderConfigs` but that method does not prioritize user secret configs which some callers might need.

Reply:
I propose to keep this as is and after Michal's improvements (https://github.com/redhat-appstudio/service-provider-integration-operator/pull/400) work on refactoring this together with the next task in token refresh.

Reply:
Adapted to changes of #400.